include/benchmark/benchmark.h

```diff
@@ -343,12 +343,24 @@ class Counter {
     kDefaults = 0,
     // Mark the counter as a rate. It will be presented divided
     // by the duration of the benchmark.
-    kIsRate = 1,
+    kIsRate = 1U << 0U,
     // Mark the counter as a thread-average quantity. It will be
     // presented divided by the number of threads.
-    kAvgThreads = 2,
+    kAvgThreads = 1U << 1U,
     // Mark the counter as a thread-average rate. See above.
-    kAvgThreadsRate = kIsRate | kAvgThreads
+    kAvgThreadsRate = kIsRate | kAvgThreads,
+    // Mark the counter as a constant value, valid/same for *every* iteration.
+    // When reporting, it will be *multiplied* by the iteration count.
+    kIsIterationInvariant = 1U << 2U,
```

Comment:
can you maybe add some realistic examples of these? i'm having a hard time thinking what an iteration invariant counter would be measuring that's useful.

Reply:
https://github.com/darktable-org/rawspeed/commit/a1ebe07bea5738f8607b48a7596c172be249590e - here, `kIsIterationInvariantRate` would be nice
https://github.com/darktable-org/rawspeed/commit/0891555be56b24f9f4af716604cedfa0da1efc6b - `kAvgIterations`

Basically, `kAvgIterations` means that the value is collected over all the iterations, e.g. a timer, thus for it to be meaningful, it needs to be divided by iteration count
And `kIsIterationInvariant` means that every iteration produces this very counter value.
You could use it as is, but if you use anything else other than `kDefaults`, the produced value may be incorrect.